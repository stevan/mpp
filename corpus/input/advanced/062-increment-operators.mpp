# Auto-increment and auto-decrement operators (unsupported)

# Postfix increment
$count++;
$index++;

# Postfix decrement
$count--;
$index--;

# Prefix increment
++$count;
++$index;

# Prefix decrement
--$count;
--$index;

# In loops
while ($i < 10) {
    print $i++;
}

for (my $i = 0; $i < 10; $i++) {
    print $i;
}

# Postfix with condition
$x++ until $x > 100;
print $line while $count++ < 10;