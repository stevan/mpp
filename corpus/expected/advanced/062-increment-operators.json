[
  {
    "type": "PostfixOp",
    "operator": "++",
    "operand": {
      "type": "Variable",
      "name": "$count"
    }
  },
  {
    "type": "PostfixOp",
    "operator": "++",
    "operand": {
      "type": "Variable",
      "name": "$index"
    }
  },
  {
    "type": "PostfixOp",
    "operator": "--",
    "operand": {
      "type": "Variable",
      "name": "$count"
    }
  },
  {
    "type": "PostfixOp",
    "operator": "--",
    "operand": {
      "type": "Variable",
      "name": "$index"
    }
  },
  {
    "type": "PrefixOp",
    "operator": "++",
    "operand": {
      "type": "Variable",
      "name": "$count"
    }
  },
  {
    "type": "PrefixOp",
    "operator": "++",
    "operand": {
      "type": "Variable",
      "name": "$index"
    }
  },
  {
    "type": "PrefixOp",
    "operator": "--",
    "operand": {
      "type": "Variable",
      "name": "$count"
    }
  },
  {
    "type": "PrefixOp",
    "operator": "--",
    "operand": {
      "type": "Variable",
      "name": "$index"
    }
  },
  {
    "type": "While",
    "condition": {
      "type": "BinaryOp",
      "operator": "<",
      "left": {
        "type": "Variable",
        "name": "$i"
      },
      "right": {
        "type": "Number",
        "value": "10"
      }
    },
    "block": [
      {
        "type": "Print",
        "arguments": [
          {
            "type": "PostfixOp",
            "operator": "++",
            "operand": {
              "type": "Variable",
              "name": "$i"
            }
          }
        ]
      }
    ]
  },
  {
    "type": "Error",
    "message": "foreach requires a scalar iterator variable",
    "value": "(",
    "line": 24,
    "column": 5
  },
  {
    "type": "BinaryOp",
    "operator": "<",
    "left": {
      "type": "Variable",
      "name": "$i"
    },
    "right": {
      "type": "Number",
      "value": "10"
    }
  },
  {
    "type": "PostfixOp",
    "operator": "++",
    "operand": {
      "type": "Variable",
      "name": "$i"
    }
  },
  {
    "type": "While",
    "condition": {
      "type": "BinaryOp",
      "operator": "<",
      "left": {
        "type": "PostfixOp",
        "operator": "++",
        "operand": {
          "type": "Variable",
          "name": "$count"
        }
      },
      "right": {
        "type": "Number",
        "value": "10"
      }
    },
    "block": [
      {
        "type": "Print",
        "arguments": [
          {
            "type": "Variable",
            "name": "$line"
          }
        ]
      }
    ]
  }
]